src/response.rs

```diff
@@ -539,9 +539,30 @@ impl FromStr for Response {
     }
 }
 
-fn read_next_line(reader: &mut impl BufRead) -> io::Result<String> {
+fn read_next_line(reader: &mut impl BufRead, context: &str) -> io::Result<String> {
     let mut s = String::new();
-    if reader.read_line(&mut s)? == 0 {
+    let result = reader.read_line(&mut s);
+
+    if let Err(e) = result {
+        // Provide context to errors encountered while reading the line.
+        // ureq does not currently handle non-ascii status lines and
+        // header values. For historical reasons, the HTTP spec does
+        // allow for characters in the range 0x80-0xff, but these are
+        // very rarely encountered in the wild.
+        // See https://github.com/algesten/ureq/issues/320
+        let reason = if e.kind() == io::ErrorKind::InvalidData {
+            format!(
+                "Invalid data in {} (see issue #320): {}",
```

Comment:
People are likely to see this error in contexts where it's not clear it comes from ureq, so `issue #320` will be confusing. I would recommend omitting it; if you'd like to include it, let's use a full URL.

Reply:
Ok!